src/internal/operators/groupBy.ts

```diff
@@ -165,20 +165,14 @@ export function groupBy<T, K, R>(
       // next call from the source.
       const handleError = (err: any) => notify((consumer) => consumer.error(err));
 
-      // The number of actively subscribed groups
-      let activeGroups = 0;
-
-      // Whether or not teardown was attempted on this subscription.
-      let teardownAttempted = false;
-
       // Capturing a reference to this, because we need a handle to it
       // in `createGroupedObservable` below. This is what we use to
       // subscribe to our source observable. This sometimes needs to be unsubscribed
       // out-of-band with our `subscriber` which is the downstream subscriber, or destination,
       // in cases where a user unsubscribes from the main resulting subscription, but
       // still has groups from this subscription subscribed and would expect values from it
       // Consider:  `source.pipe(groupBy(fn), take(2))`.
-      const groupBySourceSubscriber = new OperatorSubscriber(
+      const groupBySourceSubscriber = createOperatorSubscriber(
         subscriber,
         (value: T) => {
           // Because we have to notify all groups of any errors that occur in here,
```

Comment:
GH didn't allow me to put a comment in the exact spot lines 189-191 [here](https://github.com/ReactiveX/rxjs/pull/7252/files#diff-454b92b7ec8b4a19e5475e7e52f58f37ad6230c46835eeb02f3cbee0e32c0ac6R189-R191) but I think the comment also needs revisiting as the mentioned reference counting seem to be removed now.